eventDispatcher.go

```diff
@@ -114,70 +110,81 @@ func (d *eventDispatcher) OnDeviceEvent(event *device.Event) {
 
 	switch event.Type {
 	case device.Connect:
-		eventType, message = newOnlineMessage(d.source, event.Device)
-		url, err = d.encodeAndDispatchEvent(eventType, wrp.Msgpack, message)
+		scheme = wrp.SchemeEvent
+		eventType, message := newOnlineMessage(d.source, event.Device)
+		_, err = d.encodeAndDispatchEvent(eventType, wrp.Msgpack, message)
 		if err != nil {
 			d.logger.Error("Error dispatching online event", zap.Any("eventType", eventType), zap.Any("destination", message.Destination), zap.Error(err))
 		}
 
 	case device.Disconnect:
-		eventType, message = newOfflineMessage(d.source, event.Device)
-		url, err = d.encodeAndDispatchEvent(eventType, wrp.Msgpack, message)
+		scheme = wrp.SchemeEvent
+		eventType, message := newOfflineMessage(d.source, event.Device)
+		_, err = d.encodeAndDispatchEvent(eventType, wrp.Msgpack, message)
 		if err != nil {
 			d.logger.Error("Error dispatching offline event", zap.Any("eventType", eventType), zap.Any("destination", message.Destination), zap.Error(err))
 		}
 	case device.MessageReceived:
-		if routable, ok := event.Message.(wrp.Routable); ok {
-			destination := routable.To()
-			contentType := event.Format.ContentType()
-			if strings.HasPrefix(destination, EventPrefix) {
-				var l wrp.Locator
-				if l, err = wrp.ParseLocator(destination); err == nil {
-					eventType = l.Authority
-					url, err = d.dispatchEvent(eventType, contentType, event.Contents)
-					if err != nil {
-						d.logger.Error("Error dispatching event", zap.Any("eventType", eventType), zap.Any("destination", destination), zap.Error(err))
-					}
-				}
-			} else if strings.HasPrefix(destination, DNSPrefix) {
-				eventType = event.Type.String()
-				unfilteredURL := destination[len(DNSPrefix):]
-				url, err = d.dispatchTo(unfilteredURL, contentType, event.Contents, eventType)
-				if err != nil {
-					d.logger.Error("Error dispatching to endpoint", zap.Any("destination", destination), zap.Error(err))
-				}
-			} else {
-				eventType = event.Type.String()
-				err = ErrorUnroutableDestination
-				d.logger.Error("Unroutable destination", zap.Any("destination", destination))
-			}
+		scheme, err = d.routeMessageReceivedEvent(event)
```

Comment:
took this opportunity to make the code more readable/maintainable and reduce its cognitive load